### DISCUSSION TITLE
Account API Errors when field value is null for optional nullable field.

### DISCUSSION THREAD

**Discussion by jafin** [firefly-iii/firefly-iii#5661] [Firefly III]
I've receiving an error when using the API to create an account.

I've generated a client from the 1.5.5 firefly iii schema.
When I Send the following payload to create an account `POST /api/v1/accounts`

```json
{
	"type": "expense",
	"account_role": null,
	"credit_card_type": null,
	"liability_type": null,
	"liability_direction": null,
	"interest_period": null,
	"name": "ACCOUNT NAME",
	"iban": null,
	"bic": null,
	"account_number": null,
	"opening_balance_date": null,
	"active": true,
	"include_net_worth": true,
	"monthly_payment_date": null,
	"interest": "0",
	"notes": null,
	"latitude": null,
	"longitude": null,
	"zoom_level": null
}
```

I receive the error:

```json
{
	"message": "The given data was invalid.",
	"errors": {
		"account_role": ["The selected account role is invalid."],
		"credit_card_type": ["The selected credit card type is invalid."],
		"monthly_payment_date": ["The monthly payment date is not a valid date."],
		"liability_type": ["The selected liability type is invalid."],
		"liability_direction": ["The selected liability direction is invalid."],
		"interest_period": ["The selected interest period is invalid."]
	}
}
```

If I remove all the fields with null values, and resubmit, it will work

```json
{
	"type": "expense",
	"name": "ACCOUNT NAME",
	"active": true,
	"include_net_worth": true,
	"interest": "0"
}
```

From reading this page https://www.apimatic.io/blog/2021/09/using-optional-and-nullable-properties-in-api-requests/, 
I understand `"property": null`  is valid for an Optional nullable field.

I'm unsure if this SHOULD work with the first payload, or if it is an issue with the client generated code...?

**Comment by JC5** [comment]
That should work yes, thanks for pointing it out. I’ll pick it up.

**Reply by jafin** [reply]
Great thank you! 😄

**Reply by JC5** [reply]
The demo site should work correctly, but it's not part of a release yet.
